### PR TITLE
Add TRTemperatureComparisonFormatter+initWithDate for context aware comparisons

### PR DESCRIPTION
**Pull request by jakecraige**
There was a bug where comparisons
shown from the cache would show inaccurate text
because it wasn't using the date from the cache.

Ex:
If the date was cached in the AM and you open the app in the afternoon it might say,
"Last updated 8:00AM, It's the same this afternoon as yesterday afternoon."
when it should say,
"Last updated 8:00AM, It's the same this morning as yesterday morning."

This PR has the comparison be initialized with a date
that it uses when checking the time of day.

I chose to switch from a class method to instance method
so that I didn't have to pass the date
through a chain of class methods.

### CODE REVIEW

File: Tropos.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -198,7 +200,6 @@
 		DED70296CC99B4AAD0DE42BD /* Pods.release.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = Pods.release.xcconfig; path = "Pods/Target Support Files/Pods/Pods.release.xcconfig"; sourceTree = "<group>"; };
 		E2B218FB5F097AAE3080F4DD /* UnitTests-Prefix.pch */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = sourcecode.c.h; path = "UnitTests-Prefix.pch"; sourceTree = "<group>"; };
 		E59B090E67C8BD269D3CD9DB /* Main.storyboard */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = file.storyboard; path = Main.storyboard; sourceTree = "<group>"; };
-		E799194B1ACEE75C00A5600C /* Secrets.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = Secrets.h; sourceTree = "<group>"; };
```
Comment (tonyd256) [inline Tropos.xcodeproj/project.pbxproj]: why is this removed here?

Comment (jakecraige) [inline Tropos.xcodeproj/project.pbxproj]: It probably shouldn't. It keeps happening on my comp and I'm not sure why. I'll undo it.

**Comment by tonyd256** [comment]
@jakecraige what do you mean by "through a chain of class methods." Would it be fine to just pass the date into the class method?

File: Tropos.xcodeproj/project.pbxproj (hunk 1 above)
Comment (tonyd256) [inline Tropos.xcodeproj/project.pbxproj]: probably because it's in the .gitignore or something

**Comment by jakecraige** [comment]
@tonyd256 Because `localizedStringFromComparison:adjective:precipitation` ends up calling other class methods I would have to pass the date through the entire chain of them. 

Mainly it's because `timeOfDay` needs the date and it's being called form a few places, so I'd also have to pass in the date to those methods(`localizedCurrentTimeOfDay` and `localizedPreviousTimeOfDay`). 

They would have to accept the date and pass it through to `timeOfDay`. I could also perhaps pass in a `TRTimeOfDay` so it doesn't have to call `timeOfDay`. 

I could give it a shot like that if you think it'd be preferable.

**Comment by tonyd256** [comment]
Makes sense :+1:

**Comment by jakecraige** [comment]
Closing for #144
